### PR TITLE
fix(plugin-meetings):  Windows/Chrome still asks for camera permissions even when not selecting videoEnabled and shareVideoEnabled

### PR DESCRIPTION
**Pull request by Parimala032**
# COMPLETES #< [SPARK-539683](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-539683) >

## This pull request addresses

This pull request addresses the issue with a kitchen sink app. When the user tries to disable the videoEnabled checkbox and sharVideoEnabled checkbox, the sample app still asks for camera access. This pull request resolves this problem with the following changes:
## by making the following changes
Changes have been made in the plugin-meetings where previously getDevices() was being called without considering the user's usage. Now, getDevices is being called with parameters based on the user's requests.

https://github.com/user-attachments/assets/011647bd-81ac-4433-b1d0-6be72c12909e

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested
Tested this Pull request by 
1) uncheck the _**videoEnabled**_ and _**shareVideoEnabled**_ checkbox and click on _**joinWithMedia()**_ button ----> ask for microphone permission
2) uncheck the _**audioEnabled**_ and **_shareAudioEnabled_** checkbox and click on _**joinWithMedia()**_ button -----> ask for camera permission
3) checked both checkboxes ----> ask for the both permissions
4) Checked with the **_**mid-meeting**_** 

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 1]
```diff
@@ -6512,12 +6512,21 @@ export default class Meeting extends StatelessWebexPlugin {
...
+  private static async handleDeviceLogging(
+    isAudioEnabled = false,
+    isVideoEnabled = false
+  ): Promise<void> {
     try {
-      const devices = await getDevices();
+      const deviceKind =
+        (isVideoEnabled && Media.DeviceKind.VIDEO_INPUT) ||
+        (isAudioEnabled && Media.DeviceKind.AUDIO_INPUT) ||
+        undefined;
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:6527]: @mkesavan13 Im unable to understand this logic. When isVideoEnabled is false and isAudioENabled is false, then shouldnt deviceKind = undefined. And that should call getDevice(undfiened) -> which means we ask for both the permissions.

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 2]
```diff
@@ -4196,6 +4200,8 @@ describe('plugin-meetings', () => {
             // and that these were the only /media requests that were sent
             assert.calledTwice(locusMediaRequestStub);
             assert.calledOnce(handleDeviceLoggingSpy);
+            assert.calledWith(handleDeviceLoggingSpy, false);
+
           });
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:4206]: Please add test for the following conditions
1. Only asked for camera permission
2. Only asked for mic permission
3. Ask for both permissions
4. Ask for neither permissions.

**Review by Shreyas281299 — CHANGES_REQUESTED** [review]

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 3]
```diff
@@ -6512,12 +6512,21 @@ export default class Meeting extends StatelessWebexPlugin {
    *
    * @private
    * @static
+   * @param {boolean} isAudioEnabled
+   * @param {boolean} isVideoEnabled
    * @returns {Promise<void>}
    */
-  private static async handleDeviceLogging(): Promise<void> {
+  private static async handleDeviceLogging(
+    isAudioEnabled = false,
```
Comment (sreenara) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:6520]: Why do we need to set default values for these booleans?

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 4]
```diff
@@ -6512,12 +6512,21 @@ export default class Meeting extends StatelessWebexPlugin {
...
+    isVideoEnabled = false
+  ): Promise<void> {
     try {
-      const devices = await getDevices();
+      const deviceKind =
+        (isVideoEnabled && Media.DeviceKind.VIDEO_INPUT) ||
```
Comment (sreenara) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:6525]: Does enabling video automatically ask for audio too?
What's the reason for undefined being present here?

**Review by sreenara — CHANGES_REQUESTED** [review]
Please change the title to reflect only the change this commit is doing. The PR description can cover the rest of the details. This title is quite long.

> This pull request addresses the issue with a kitchen sink app.

Isn't this an issue with the web client too?

> Changes have been made in the plugin-meetings 

Could we be more specific about what the changes were inside plugin meetings? I don't see anything about the function that actually had the problem.

Please also add a root cause to the PR description.

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 1 above)
Comment (mkesavan13) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:6527]: @Parimala032 can answer to this. Logically that is how it is supposed to work, yes

**Review by mkesavan13 — CHANGES_REQUESTED** [review]
Please address @sreenara & @Shreyas281299 comments. Meanwhile, we can have tests written and demo recorded for following cases as well,
- Video is enabled and Audio disabled
- Both video and audio enabled
- Both video and audio disabled

**Comment by Parimala032** [comment]
Closing this PR where we need to add the UT cases properly and there are more changes that need to include regarding the UT will raise another PR

**Pull request closed without merging**